cmd/juju/application/deployer/bundle.go

```diff
@@ -112,9 +114,17 @@ Please repeat the deploy command with the --trust argument if you consent to tru
 					return errors.Trace(err)
 				}
 
+				origin, err := utils.DeduceOrigin(charmURL, corecharm.Channel{})
```

Comment:
Can't you specify a channel to use for an app in the bundle syntax?  Use that here?